lib/dry/types/default.rb

```diff
@@ -23,7 +24,7 @@ def evaluate
       alias_method :evaluate, :value
 
       # @param [Object, #call] value
-      # @return [Default, Dry::Types::Default::Callable]
+      # @return [Class] `Default` or `Callable`
```

Comment:
When I wrote this, I have tried to make documentation more helpful and provide links to returned classes.
Unfortunately, AFAIK [YARD](http://yardoc.org/) does not support putting specific `Class` instances returned from a method in `@return` tag (because the class name in `@return` block means “instance of the class”). 

Following that direction, to have links to those classes, I think, It is probably better to link to the actual classes, and not only mark them as code in `@return` so following would be more helpful:

```ruby
# @return [Class] {Default} or {Default::Callable}
```